models/Import/Service/ResultImporter.php

```diff
@@ -0,0 +1,369 @@
+        $testScoreVariables = $this->getTestScoreVariables($resultStorage, $deliveryExecutionUri);
+
+        $resultStorage->getPersistence()->transactional(
+            function () use ($resultStorage, $input, $testUri, $deliveryExecutionUri, $testScoreVariables) {
+                $this->updateItemResponseVariables($resultStorage, $input, $testUri);
+
+                $updatedScoreTotal = $this->updateItemOutcomeVariables(
+                    $resultStorage,
+                    $input,
+                    $testUri,
+                    $testScoreVariables['scoreTotalMax'],
+                    $testScoreVariables['updatedScoreTotal']
+                );
+
+                $this->updateTestVariables(
+                    $resultStorage,
+                    $testScoreVariables['scoreTotalVariable'],
+                    $testScoreVariables['scoreTotalVariableId'],
+                    $deliveryExecutionUri,
+                    $testUri,
+                    $updatedScoreTotal
+                );
+            }
+        );
+    }
+
+    private function updateTestVariables(
+        AbstractRdsResultStorage $resultStorage,
+        taoResultServer_models_classes_Variable $scoreTotalVariable,
+        int $scoreTotalVariableId,
+        string $deliveryExecutionUri,
+        string $testUri,
+        float $updatedScoreTotal
+    ): void {
+        $scoreTotalVariable->setValue($updatedScoreTotal);
+
+        $resultStorage->replaceTestVariables(
+            $deliveryExecutionUri,
+            $testUri,
+            $deliveryExecutionUri,
+            [
+                $scoreTotalVariableId => $scoreTotalVariable
+            ]
+        );
+    }
+
+    private function updateItemResponseVariables(
+        AbstractRdsResultStorage $resultStorage,
+        ImportResultInput $input,
+        string $testUri
+    ): void {
+        $deliveryExecutionUri = $input->getDeliveryExecutionId();
+
+        foreach ($input->getResponses() as $itemId => $responses) {
+            $callItemId = $this->createCallItemId($deliveryExecutionUri, $itemId);
+            $itemVariables = [];
+
+            foreach ($responses as $responseId => $responseValue) {
+                if (!array_key_exists('correctResponse', $responseValue)) {
+                    continue;
+                }
+
+                $responseVariable = $this->getItemVariable(
+                    $resultStorage,
+                    $deliveryExecutionUri,
+                    $itemId,
+                    $callItemId,
+                    $responseId
+                );
+
+                /** @var taoResultServer_models_classes_ResponseVariable $variable */
+                $variable = $responseVariable['variable'];
+                $variableId = $responseVariable['variableId'];
+                $itemUri = $responseVariable['itemUri'];
+
+                $variable->setCorrectResponse(boolval($responseValue['correctResponse']));
+
+                $itemVariables[$variableId] = $variable;
```

Comment:
Don't we need to add `break` at the end of this `foreach`?
If `$responseValue['correctResponse']` will be at the beginning of the array - we will check for `!array_key_exists('correctResponse', $responseValue)` multiple times. Not critical, but we can avoid calling this one multiple times 🙂